libtenzir/builtins/operators/load_tcp.cpp

```diff
@@ -333,11 +333,13 @@ struct connection_manager_state {
       caf::anon_send(metrics_receiver, operator_id, tcp_metrics_id,
                      std::move(metric));
       if (self) {
-        next_emit_metrics = detail::weak_run_delayed(
-          self, defaults::metrics_interval,
-          [self, connection = this->shared_from_this()] {
-            connection->emit_metrics(self);
-          });
+        next_emit_metrics
+          = detail::weak_run_delayed(self, defaults::metrics_interval,
+                                     [self, weak_ptr = this->weak_from_this()] {
+                                       if (auto connection = weak_ptr.lock()) {
+                                         connection->emit_metrics(self);
+                                       }
+                                     });
```

Comment:
Somebody should double check this (also w.r.t. to the lifetime of `self`).

This fixes the problem for us that stopping and starting a managed pipeline makes it seem like no more data is accepted. This happens because the connection stays alive, as you can see above.

Reply:
This looks very much like the correct fix to an actual problem to me.